video/out/wayland_common.c

```diff
@@ -1549,6 +1549,16 @@ static const struct wl_registry_listener registry_listener = {
 };
 
 /* Static functions */
+static void free_dnd_data(struct vo_wayland_state *wl)
+{
+    if (wl->dnd_action >= 0 && wl->dnd_offer)
+        wl_data_offer_finish(wl->dnd_offer);
```

Comment:
`wl_data_offer_finish` isn't actually a freeing/resource destroying command. Not sure you want this in here.

Reply:
wasn't sure about this, however if I disconnect resetting `dnd_action` from the `finish` call there's potential for it not being called.
is that a problem?

Reply:
It's a protocol error to call it at the wrong time. Kind of hard to imagine how exactly that would happen with these changes, but AFAIK you only need it after successfully reading the data.

Reply:
moved it.